docs/content/concepts/assets/asset-checks/subsetting-asset-checks.mdx

```diff
@@ -0,0 +1,115 @@
+
+In this guide, we'll show you a few approaches to subsetting your asset checks.
+
+---
+
+## Prerequisites
+
+Before continuing, you should be familiar with:
+
+- [Asset checks](/concepts/assets/asset-checks)
+- [Multi-assets](/concepts/assets/multi-assets)
+
+---
+
+## Executing subsets of checks using @multi_asset_check
+
+Using the <PyObject object="multi_asset_check" decorator /> decorator's `specs` and `can_subset` arguments, you can execute a subset of checks in a single op.
+
+Inside the body of the function, we can use <PyObject object="AssetCheckExecutionContext" display="AssetCheckExecutionContext.selected_asset_check_keys" /> to identify which computations to run. We can also set the decorator's `can_subset` parameter to `True` to allow the computation to compute an arbitrary subset of the asset checks it contains.
+
+Setting `can_subset` to `True` means that not all asset checks may be computed when the function runs. As we don't know in advance which checks will be executed, we explicitly `yield` each asset check result that we're expected to create:
+
+```python file=/concepts/assets/asset_checks/subset_multi_asset_check.py
+import random
+from typing import Iterable
+
+from dagster import (
+    AssetCheckExecutionContext,
+    AssetCheckResult,
+    AssetCheckSeverity,
+    AssetCheckSpec,
+    multi_asset_check,
+)
+
+
+@multi_asset_check(
+    specs=[
+        AssetCheckSpec(name="asset_check_one", asset="my_asset_one"),
+        AssetCheckSpec(name="asset_check_two", asset="my_asset_two"),
+    ],
+    can_subset=True,
+)
+def the_check(context: AssetCheckExecutionContext) -> Iterable[AssetCheckResult]:
+    for check_key in context.selected_asset_check_keys:
+        asset_key = check_key.asset_key
+
+        # TODO: populate metadata
+        metadata = {}
+        if random.random() > 0.5:
+            yield AssetCheckResult(
+                passed=False,
+                severity=AssetCheckSeverity.WARN,
+                description="The asset is over 0.5",
+                asset_key=asset_key,
+            )
+        else:
+            yield AssetCheckResult(
+                passed=True,
+                metadata=metadata,
+                description="The asset is fresh.",
+                asset_key=asset_key,
+            )
+```
+
+---
+
+## TODO: With multi_asset
+
+When using [multi-assets](/concepts/assets/multi-assets), Dagster assumes that all checks specified on the asset should be executed after it is materialized. This means that attempting to execute some, but not all, of the checks defined by a multi-asset will result in an error.
+
+```python
+TODO - EXAMPLE
```

Comment:
Example needed

Reply:
here's sample code: https://github.com/dagster-io/dagster/blob/master/python_modules/dagster-test/dagster_test/toys/asset_checks.py#L140-L157

Reply:
Should we use this? In some of the de-emphasizing I/O manager work I did, we opted to move to using `AssetSpec` instead of `AssetOut`, and `yield MaterializeResult` instead of `yield Output`. 

I'm not sure if that's relevant to this specific example - it's discussed in DOC-110 - but just wanted to check so we can be consistent in the examples we're using.